src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -2319,6 +2335,21 @@ int fmi3_xml_handle_ModelVariables(fmi3_xml_parser_context_t *context, const cha
             }
         }
 
+        yyscan_t scanner;
+        YY_BUFFER_STATE buf;
+        /* check variable name syntax */
+        if (md->namingConvension == fmi3_naming_enu_structured) {
+            yyfmi3lex_init(&scanner);
+            for (size_t i = 0; i < nVars; ++i) {
+                char *name = ((fmi3_xml_variable_t *) jm_vector_get_item(jm_voidp)(
```

Comment:
What are we doing here?

Reply:
This verifies naming conventions are being followed, see [https://fmi-standard.org/docs/3.0/#namingSection](https://fmi-standard.org/docs/3.0/#namingSection). The code itself is not new, but only moved from fmi3_xml_query.

Reply:
ok